mlir/lib/Conversion/GPUCommon/OpToFuncCallLowering.h

```diff
@@ -58,7 +59,8 @@ struct OpToFuncCallLowering : public ConvertOpToLLVMPattern<SourceOp> {
         "expected single result op");
 
     static_assert(std::is_base_of<OpTrait::SameOperandsAndResultType<SourceOp>,
-                                  SourceOp>::value,
+                                  SourceOp>::value ||
+                      std::is_same_v<SourceOp, math::FPowIOp>,
```

Comment:
This feels like a hack. Are there some other things we could check instead? Maybe turn this into a runtime assertion and check that the first operand and have the same type, or add a hook for getting the element types with a default for `SameOperandsAndResultType`?

Reply:
>  Maybe turn this into a runtime assertion and check that the first operand and have the same type

Yeah it looks like a hack, but honestly, isn't checking only the first operand relaxes even more of the static_assert checks? 

Well I cannot think of a better way to do it so I just went with your suggestion to check the first operand.